app/commands/track/update_build_status.rb

```diff
@@ -0,0 +1,359 @@
+class Track::UpdateBuildStatus
+  include Mandate
+
+  initialize_with :track
+
+  def call
+    track.build_status = build_status
+  end
+
+  private
+  def build_status
+    {
+      health:,
+      volunteers:,
+      students:,
+      submissions:,
+      mentor_discussions:,
+      syllabus:,
+      practice_exercises:,
+      test_runner:,
+      representer:,
+      analyzer:
+    }
+  end
+
+  def health
+    num_exemplar = component_health_statuses[:exemplar].to_i
+    num_healthy = component_health_statuses[:healthy].to_i
+
+    return :exemplar if num_exemplar == 5
+    return :healthy if num_exemplar + num_healthy == 5
+
+    :needs_attention
```

Comment:
I'm being quite strict, where the overall status is `:needs_attention` when at least one component has the `:needs_attention` or `:missing` status.